grpc/server/admin_test.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	adminv1 "github.com/nanzhong/tstr/api/admin/v1"
+	commonv1 "github.com/nanzhong/tstr/api/common/v1"
+	"github.com/nanzhong/tstr/db"
+	"github.com/nanzhong/tstr/grpc/auth"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func newTestAdminServer(t *testing.T) (*AdminServer, *db.MockQuerier) {
+	ctrl := gomock.NewController(t)
+	t.Cleanup(ctrl.Finish)
+	mockQuerier := db.NewMockQuerier(ctrl)
+
+	return &AdminServer{
+		dbQuerier: mockQuerier,
+	}, mockQuerier
+}
+
+func TestAdminServer_GetAccessToken(t *testing.T) {
+	tests := []struct {
+		name              string
+		token             *commonv1.AccessToken
+		errCode           codes.Code
+		errMsg            string
+		mockQuerierReturn func(*commonv1.AccessToken) (db.GetAccessTokenRow, error)
+	}{
+		{
+			name:    "valid acces token uuid in the request",
+			token:   NewAccessTokenBuilder().WithRevokedAt().Build(),
+			errCode: codes.OK,
+			errMsg:  "",
+			mockQuerierReturn: func(token *commonv1.AccessToken) (db.GetAccessTokenRow, error) {
+				tokenID, _ := uuid.Parse(token.Id)
+				return db.GetAccessTokenRow{
+					ID:                 tokenID,
+					Name:               token.Name,
+					NamespaceSelectors: token.NamespaceSelectors,
+					Scopes:             []string{"admin"},
+					IssuedAt:           sql.NullTime{Valid: true, Time: token.IssuedAt.AsTime()},
+					ExpiresAt:          sql.NullTime{Valid: true, Time: token.ExpiresAt.AsTime()},
+					RevokedAt:          sql.NullTime{Valid: true, Time: token.RevokedAt.AsTime()},
```

Comment:
Even though this doesn't affect the test, this doesn't quite line up with how the sql null types work. The `Valid` field here is used to differentiate a null sql value vs a non-null value. You can use https://github.com/nanzhong/tstr/blob/main/grpc/types/from.go#L61 to handle the conversion.

Reply:
ack! Updated!